src/analysis/souffle/tests/arcs_fact_tests/dynamic_rule_test.dl

```diff
@@ -0,0 +1,71 @@
+#include "taint.dl"
+#include "fact_test_helper.dl"
+#include "integrity_tag_prop_helper.dl"
+
+// This file explores a simple policy that attempts to add integrity tags based
+// upon the presence of other integrity tags. It has the following rules:
+//
+// 1. When a value tagged with TimestampMillis is divided by a value tagged as
+//    being the constant 1000, the result is TimestampSeconds
+//
+// 2. When a value tagged with TimestampSeconds is multiplied by a value tagged
+//    as being the constant 1000, the result is TimestampMillisRedacted.
+//
+// While we might use something like operators in the future, we are currently
+// doing something a bit more ad-hoc for this test by indicating that certain
+// access paths are the result of operation expressions with the isOperatorExpr
+// relation.
+
+.decl mult_op(result: AccessPath, operand1: AccessPath, operand2: AccessPath)
+.decl div_op(result: AccessPath, dividend: AccessPath, divisor: AccessPath)
+
+edge(op1, result), edge(op2, result) :- mult_op(result, op1, op2).
+edge(dividend, result), edge(divisor, result) :- div_op(result, dividend, divisor).
```

Comment:
Those rules handle the propagation via policy rules. However, without a deeper refactoring, automatic `IntegrityTag` and `ConfidentialityTag` propagation occurs only via `edge`s. Unless we wished to change the semantics of `edge`s to be not just "data flows between these two `AccessPath`s" to "an assignment occurs between these two `AccessPath`s (as we did in the `phi` node PR), we need this to preserve default propagation.

Reply:
Isn't this similar to the handling of `derives from` claims? Recall that we remove some intra-particle edges between inputs and outputs, when there are `derives from` claims because we should not do default propagation along some edges. Similarly, if we understand the semantics of some operation, we should use more specific propagation rules that are aligned with the semantics. no?

Reply:
Sure, I think that makes sense. In that case, getting rid of these `edge` rules would be the right thing.